UCT: Compilation failure on system without Infiniband

Github HEAD master (as of a few minutes ago)

```
checking whether ibv_mlx5_exp_get_qp_info is declared... no
checking whether ibv_mlx5_exp_get_cq_info is declared... no
checking whether ibv_mlx5_exp_get_srq_info is declared... no
checking whether ibv_mlx5_exp_update_cq_ci is declared... no
checking whether MLX5_WQE_CTRL_SOLICITED is declared... no
checking whether IBV_EXP_QP_INIT_ATTR_RES_DOMAIN is declared... no
configure: WARNING: Cannot use mlx5 accel because resource domains are not supported
configure: WARNING: Please upgrade MellanoxOFED to 3.1 or above
checking whether IBV_EXP_RES_DOMAIN_THREAD_MODEL is declared... no
configure: WARNING: Cannot use mlx5 accel because resource domains are not supported
configure: WARNING: Please upgrade MellanoxOFED to 3.1 or above
checking whether ibv_exp_create_res_domain is declared... no
configure: WARNING: Cannot use mlx5 accel because resource domains are not supported
configure: WARNING: Please upgrade MellanoxOFED to 3.1 or above
checking whether ibv_exp_destroy_res_domain is declared... no
configure: WARNING: Cannot use mlx5 accel because resource domains are not supported
configure: WARNING: Please upgrade MellanoxOFED to 3.1 or above
```

```
Making all in src/ucs
make[2]: Entering directory '/home/tc/src/ucx/build/src/ucs'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/tc/src/ucx/build/src/ucs'
Making all in src/uct
make[2]: Entering directory '/home/tc/src/ucx/build/src/uct'
/bin/sh ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../../ucx-git/src/uct -I../..  -DCPU_FLAGS="" -I/home/tc/src/ucx/build/../ucx-git/src -I/home/tc/src/ucx/build/src    -I/opt/xpmem/include  -O3 -g -Wall -Werror  -O2 -ggdb -funwind-tables -MT ib/cm/libuct_la-cm_iface.lo -MD -MP -MF ib/cm/.deps/libuct_la-cm_iface.Tpo -c -o ib/cm/libuct_la-cm_iface.lo `test -f 'ib/cm/cm_iface.c' || echo '../../../ucx-git/src/uct/'`ib/cm/cm_iface.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../../ucx-git/src/uct -I../.. -DCPU_FLAGS= -I/home/tc/src/ucx/build/../ucx-git/src -I/home/tc/src/ucx/build/src -I/opt/xpmem/include -O3 -g -Wall -Werror -O2 -ggdb -funwind-tables -MT ib/cm/libuct_la-cm_iface.lo -MD -MP -MF ib/cm/.deps/libuct_la-cm_iface.Tpo -c ../../../ucx-git/src/uct/ib/cm/cm_iface.c  -fPIC -DPIC -o ib/cm/.libs/libuct_la-cm_iface.o
../../../ucx-git/src/uct/ib/cm/cm_iface.c:15:10: fatal error: infiniband/driver.h: No such file or directory
 #include <infiniband/driver.h>
          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

--- a/src/uct/ib/cm/cm_iface.c
+++ b/src/uct/ib/cm/cm_iface.c
@@ -12,7 +12,6 @@
 #include <ucs/arch/atomic.h>
 #include <ucs/async/async.h>
 #include <ucs/debug/log.h>
-#include <infiniband/driver.h>
 #include <poll.h>
 
 
@@ -459,18 +458,6 @@ static ucs_status_t uct_cm_query_resources(uct_md_h md,
                                            uct_tl_resource_desc_t **resources_p,
                                            unsigned *num_resources_p)
 {
-    static const char *cm_sysfs_name = "class/infiniband_cm/abi_version";
-    char value[8];
-
-    if (ibv_read_sysfs_file(ibv_get_sysfs_path(), cm_sysfs_name,
-                            value, sizeof value) < 0) {
-        ucs_debug("failed to open %s/%s, disabling cm transport",
-                  ibv_get_sysfs_path(), cm_sysfs_name);
-        *num_resources_p = 0;
-        *resources_p     = NULL;
-        return UCS_OK;
-    }
-
     return uct_ib_device_query_tl_resources(&ucs_derived_of(md, uct_ib_md_t)->dev,
                                             "cm", UCT_IB_DEVICE_FLAG_LINK_IB,
                                             resources_p, num_resources_p);

